### PR TITLE
[CBRD-23513] fix memory leak of untreated error in db_string_pad

### PR DESCRIPTION
**Pull request by ribram**
http://jira.cubrid.org/browse/CBRD-23513

"Catch" bad precision error in db_string_pad. qstr_make_typed_string does not detect such errors.

Added a safe-guard to qstr_make_typed_string to make sure no error is missed.

### CODE REVIEW

**Comment by cubridci** [comment]
Build comment file: 
--------------
**
** This Pull Request's title is 'fix memory leak of untreated error in db_string_pad'.
** And it does not start with a related JIRA issue number **
** Please edit this Pull Request's title. (eg: '[CBRD-20309] Redesign vacuum data flush' **
**

--------------

File: src/query/string_opfunc.c [hunk 1]
```diff
@@ -3992,6 +3992,16 @@ db_string_pad (const MISC_OPERAND pad_operand, const DB_VALUE * src_string, cons
 			   db_get_string_length (src_string), db_get_string_size (src_string),
 			   db_get_string_codeset (src_string), &result, &result_type, &result_length, &result_size);
 
+  int max_size = MIN ((int) prm_get_bigint_value (PRM_ID_STRING_MAX_SIZE_BYTES),
```
Comment (razvanradulescu) [inline src/query/string_opfunc.c]: result_length = length in chars, should be compared to precision
result_size= size in bytes, should be compared with prm_get_bigint_value (PRM_ID_STRING_MAX_SIZE_BYTES)

Comment (ribram) [reply src/query/string_opfunc.c]: fixed

**Comment by ribram** [comment]
@eseokoh 

It might be best to trigger manual test first. qstr_make_typed_string safe-guards may cause many crashes.

**Pull request merged**